cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -212,9 +223,68 @@ func (ng *nodegroup) Nodes() ([]cloudprovider.Instance, error) {
 // fully populated Node object, with all of the labels, capacity and
 // allocatable information as well as all pods that are started on the
 // node by default, using manifest (most likely only kube-proxy).
-// Implementation optional.
 func (ng *nodegroup) TemplateNodeInfo() (*schedulernodeinfo.NodeInfo, error) {
-	return nil, cloudprovider.ErrNotImplemented
+	if !ng.scalableResource.CanScaleFromZero() {
+		return nil, cloudprovider.ErrNotImplemented
+	}
+
+	cpu, err := ng.scalableResource.InstanceCPUCapacity()
+	if err != nil {
+		return nil, err
+	}
+
+	mem, err := ng.scalableResource.InstanceMemoryCapacity()
+	if err != nil {
+		return nil, err
+	}
+
+	gpu, err := ng.scalableResource.InstanceGPUCapacity()
+	if err != nil {
+		return nil, err
+	}
+
+	pod, err := ng.scalableResource.InstanceMaxPodsCapacity()
+	if err != nil {
+		return nil, err
+	}
+
+	if cpu.IsZero() || mem.IsZero() {
+		return nil, cloudprovider.ErrNotImplemented
+	}
+
+	if gpu.IsZero() {
+		gpu = zeroQuantity.DeepCopy()
+	}
+
+	if pod.IsZero() {
+		pod = *resource.NewQuantity(defaultMaxPods, resource.DecimalSI)
+	}
+
+	capacity := map[corev1.ResourceName]resource.Quantity{
+		corev1.ResourceCPU:        cpu,
+		corev1.ResourceMemory:     mem,
+		corev1.ResourcePods:       pod,
+		gpuapis.ResourceNvidiaGPU: gpu,
+	}
+
+	nodeName := fmt.Sprintf("%s-asg-%d", ng.Name(), rand.Int63())
```

Comment:
Where does `asg` come from in this? 🤔 Sounds like an AWS term to me 😅

Reply:
good question, i'll work on find a doc link or something

Reply:
ok, dug in a little bit.

i have a feeling that "asg" is inspired by [AutoScalingGroups](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html) from Amazon.

and indeed, this is how it is implemented in the aws provider:
https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/aws_manager.go#L350

but, and probably due to copying by other vendors this also appears in:
AliCloud:
https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/alicloud/alicloud_manager.go#L212

Azure:
https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go#L517

BaiduCloud:
https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/baiducloud/baiducloud_manager.go#L244

i suspect all these implementations are inspired by the aws provider, as was ours.

GCE for example uses `-template-`,
https://github.com/openshift/kubernetes-autoscaler/blob/master/cluster-autoscaler/cloudprovider/gce/templates.go#L126

i'm not sure if there is a "correct" answer here, but i am leaning towards leaving this the way it is unless we have a clear idea of what we would prefer. i am open to suggestions.